src/test/ui/issues/issue-54954.stderr

```diff
@@ -4,14 +4,19 @@ error[E0379]: trait fns cannot be declared const
 LL |     const fn const_val<T: Sized>() -> usize {
    |     ^^^^^ trait fns cannot be const
 
-error[E0283]: type annotations needed: cannot resolve `_: Tt`
+error[E0283]: type annotations needed
   --> $DIR/issue-54954.rs:3:24
    |
 LL | const ARR_LEN: usize = Tt::const_val::<[i8; 123]>();
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |                        |
+   |                        cannot infer type
+   |                        help: consider specifying the type argument in the function call: `Tt::const_val::<[i8; 123]>::<T>`
```

Comment:
This output is incorrect, but it only affects a nightly-only feature, I'm tempted at letting this slide for now as we have no way to deal with this at the moment (we don't have a `PathSegment` in `traits/error_reporting.rs`). I'd be inclined at merging this as is, given how many other more common cases are improved by this PR. This test should probably be handled more gracefully in general.

I can also do snippet checks, which I would prefer to avoid.